templates/flutter/lib/src/client_offline_mixin.dart.twig

```diff
@@ -0,0 +1,660 @@
+import 'package:sembast/timestamp.dart';
+import 'package:sembast/utils/value_utils.dart';
+
+import 'enums.dart';
+import 'exception.dart';
+import 'offline_db.dart';
+import 'response.dart';
+
+class AccessTimestamp {
+  final String model;
+  final String key;
+  final Timestamp accessedAt;
+
+  AccessTimestamp({
+    required this.model,
+    required this.key,
+    required this.accessedAt,
+  });
+
+  factory AccessTimestamp.fromMap(Map<String, Object?> json) => AccessTimestamp(
+        model: json["model"] as String,
+        key: json["key"] as String,
+        accessedAt: json["accessedAt"] as Timestamp,
+      );
+
+  Map<String, Object?> toMap() => {
+        "model": model,
+        "key": key,
+        "accessedAt": accessedAt,
+      };
+}
+
+class ClientOfflineMixin {
+  static const defaultLimit = 25;
+  ValueNotifier<bool> isOnline = ValueNotifier(true);
+  late Database db;
+
+  StoreRef<String, Map<String, Object?>> _queuedWritesStore =
+      stringMapStoreFactory.store('queuedWrites');
+  StoreRef<String, Map<String, Object?>> _accessTimestampsStore =
+      stringMapStoreFactory.store('accessTimestamps');
+  StoreRef<String, int> _cacheSizeStore = StoreRef<String, int>('cacheSize');
+
+  Future<void> initOffline({
+    required Future<Response<dynamic>> Function(
+      HttpMethod, {
+      String path,
+      Map<String, String> headers,
+      Map<String, dynamic> params,
+      ResponseType? responseType,
+      String cacheModel,
+      String cacheKey,
+      String cacheResponseIdKey,
+      String cacheResponseContainerKey,
+    })
+        call,
+    void Function(Object)? onWriteQueueError,
+    required int Function() getOfflineCacheSize,
+  }) async {
+    await Future.wait([initOfflineDatabase(), listenForConnectivity()]);
+    await processWriteQueue(call, onError: onWriteQueueError);
+
+    final cacheSizeRecordRef = getCacheSizeRecordRef();
+    cacheSizeRecordRef.onSnapshot(db).listen((snapshot) {
+      int? currentSize = snapshot?.value;
+
+      if (currentSize == null || currentSize < getOfflineCacheSize()) return;
+
+      db.transaction((txn) async {
+        final records = await listAccessedAt(txn);
+        if (records.isEmpty) return;
+        final record = records.first;
+        final modelStore = getModelStore(record.value['model'] as String);
+        final cacheKey = record.value['key'] as String;
+        await deleteCache(txn, modelStore, key: cacheKey);
+      });
+    });
+  }
+
+  Future<void> initOfflineDatabase() async {
+    db = await OfflineDatabase.instance.db();
+  }
+
+  Future<void> processWriteQueue(
+      Future<Response<dynamic>> Function(
+    HttpMethod, {
+    String path,
+    Map<String, String> headers,
+    Map<String, dynamic> params,
+    ResponseType? responseType,
+    String cacheModel,
+    String cacheKey,
+    String cacheResponseIdKey,
+    String cacheResponseContainerKey,
+  })
+          call,
+      {void Function(Object e)? onError}) async {
```

Comment:
The indentation looks a little off here

Reply:
This is actually how the dart formatter auto-formats this.